navigator/runtime/src/main/java/com/freeletics/mad/navigator/NavEventNavigator.kt

```diff
@@ -90,6 +93,34 @@ public open class NavEventNavigator {
         return request
     }
 
+    /**
+     * Register for receiving navigation results that were delivered through
+     * [deliverNavigationResult]. [T] is expected to be the [BaseRoute] to the current destination.
+     *
+     * The returned [NavigationResultRequest] has a [NavigationResultRequest.Key]. This `key` should
+     * be passed to the target destination which can then use it to call [deliverNavigationResult].
+     *
+     * Note: This has to be called *before* this [NavEventNavigator] gets attached to a fragment.
+     *   In practice, this means it should usually be called during initialisation of your subclass.
+     */
+    public inline fun <reified T : BaseRoute, reified O : Parcelable> registerForNavigationResult():
+        NavigationResultRequest<O> {
+        return registerForNavigationResult(T::class, O::class)
+    }
+
+    @InternalNavigatorApi
+    public fun <O : Parcelable> registerForNavigationResult(
+        route: KClass<out BaseRoute>,
+        result: KClass<*>
+    ): NavigationResultRequest<O> {
+        checkAllowedToAddRequests()
+        val requestKey = route.qualifiedName!! + result.qualifiedName!!
```

Comment:
What do you think about adding `.` in between for better readability in case this would be debugged?

Reply:
Sounds like a good idea, using a `-` though because the qualified name already has dots in it.
```suggestion
        val requestKey = "${route.qualifiedName!!}-${result.qualifiedName!!}"
```